Unify blob store compress setting

henningandersen opened the pull request:
Blob store compression was all implemented generally, except reading the
setting for it. Moved the setting to BlobStoreRepository to unify this.
Also removed deprecated env setting 'repositories.fs.compress'.

This is a follow up on #39073

Given this is not documented, does this need to be included in the breaking changes
documentation for 8.0?

henningandersen committed:
Unify blob store compress setting

Blob store compression was all implemented generally, except reading the
setting for it. Moved the setting to BlobStoreRepository to unify this.
Also removed deprecated env setting 'repositories.fs.compress'.

This is a follow up on #39073

elasticmachine commented:
Pinging @elastic/es-distributed

original-brownbear commented on server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryTests.java:
```diff
@@ -248,40 +247,28 @@ public void testBadChunksize() throws Exception {
-        new FsRepository(metaData, useCompressEnvironment, null);
-
-        assertWarnings("[repositories.fs.compress] setting was deprecated in Elasticsearch and will be removed in a future release!" +
-            " See the breaking changes documentation for the next major version.");
```
@henningandersen maybe we should add a small note, given that we kept logging this?

henningandersen replied:
Yes, makes sense, thanks.

original-brownbear reviewed (approved):
LGTM :)

Not sure about adding something to the breaking changes (instinctively it may be a good idea, given the previous log message) => @ywelsch wdyt?

ywelsch reviewed (approved):
LGTM. I'm inclined to add a note to the breaking changes docs, as I'm not sure exactly for what purpose this was added and who might be using it. @imotov @dadoonet do you remember where this undocumented node setting `repositories.fs.compress` is coming from?

henningandersen committed:
Unify blob store compress setting

Added breaking changes documentation on the removed
'repositories.fs.compress' setting.

This is a follow up on #39073

henningandersen commented:
I added the breaking changes documentation for the removed setting.

henningandersen commented:
@elasticmachine run elasticsearch-ci/bwc
@elasticmachine run CLA

henningandersen committed:
Merge remote-tracking branch 'origin/master' into fix_blob_store_unify_compress
